lightning/src/onion_message/messenger.rs

```diff
@@ -926,6 +946,24 @@ where
 			.map_err(|_| SendError::PathNotFound)
 	}
 
+	fn create_blinded_path(&self) -> Result<BlindedPath, SendError> {
+		let recipient = self.node_signer
+			.get_node_id(Recipient::Node)
+			.map_err(|_| SendError::GetNodeIdFailed)?;
+		let secp_ctx = &self.secp_ctx;
+
+		let peers = self.message_recipients.lock().unwrap()
+			.iter()
+			.filter(|(_, peer)| matches!(peer, OnionMessageRecipient::ConnectedPeer(_)))
+			.map(|(node_id, _)| *node_id)
+			.collect();
+
+		self.message_router
+			.create_blinded_paths(recipient, peers, secp_ctx)
+			.and_then(|paths| paths.into_iter().next().ok_or(()))
+			.map_err(|_| SendError::PathNotFound)
```

Comment:
Seems like when this is used with `handle_onion_message_response`, `SendError::PathNotFound` may returned in two different scenarios:
- when creating a reply path (here)
- when finding a path to send the response

The docs on `SendError::PathNotFound` indicate the latter scenario. I'm not sure if differentiating the two is that important, though, since in the latter case we log. Maybe we should log in both cases?

Reply:
I think logging for the failed `reply_path` case would be a reasonable thing to do!
Introduce a logging mechanism in the `handle_onion_message_response` for it in [8d5f9dc](https://github.com/lightningdevkit/rust-lightning/commit/8d5f9dce49b9b2b49a2e67d52a3021325ac3fcb3)
Let me know if that looks good!
Thanks for the suggestion!

Reply:
Let's also update the docs on `SendError::PathNotFound`.

Reply:
Updated the `SendError::PathNotFound` docs in [27899b1](https://github.com/lightningdevkit/rust-lightning/commit/27899b1a04bfd78db34f2c22ad489f1a6fd4e5e4)
Let me know if they look apt!
Thanks for the suggestion!